rpcserver.go

```diff
@@ -6748,6 +6754,76 @@ func (r *rpcServer) ListPermissions(_ context.Context,
 	}, nil
 }
 
+// ListPermissions lists all RPC method URIs and their required macaroon
+// permissions to access them.
+func (r *rpcServer) UpdateNodeAnnouncement(_ context.Context,
+	req *lnrpc.NodeAnnouncementUpdateRequest) (
+	*lnrpc.NodeAnnouncementUpdateResponse, error) {
+
+	nodeModifiers := make([]netann.NodeAnnModifier, 0)
+
+	// TODO(positiveblue): add Features modifier
+
+	if req.Color != "" {
+		color, err := parseHexColor(req.Color)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse color: %v", err)
+		}
+
+		nodeModifiers = append(nodeModifiers, netann.NodeAnnSetColor(color))
+	}
+
+	if req.Alias != "" {
+		alias, err := lnwire.NewNodeAlias(req.Alias)
+		if err != nil {
+			return nil, fmt.Errorf("invalid alias value: %v", err)
+		}
+		nodeModifiers = append(nodeModifiers, netann.NodeAnnSetAlias(alias))
+	}
+
+	if len(req.Addresses) > 0 {
+		// TODO(positiveblue): add address validation
+		var newAddrs []net.Addr
+
+		for _, addr := range req.Addresses {
+			newAddr, err := net.ResolveTCPAddr(addr.Network, addr.Addr)
+			if err != nil {
+				return nil, fmt.Errorf("unable to resolve "+
+					"address %v: %v", addr, err)
+			}
+			newAddrs = append(newAddrs, newAddr)
+		}
+
+		nodeModifiers = append(nodeModifiers, netann.NodeAnnSetAddrs(newAddrs))
```

Comment:
IIRC, this'll result in all the new addresses _replacing_ what ever we had advertised, but I think we want to instead either remove/replace the address (so mutate what we already advertise).